homeassistant/components/group/__init__.py

```diff
@@ -367,6 +368,15 @@ async def groups_service_handler(service: ServiceCall) -> None:
                 entity_ids = set(group.tracking) | set(delta)
                 await group.async_update_tracked_entity_ids(entity_ids)
 
+            if ATTR_REMOVE_ENTITIES in service.data:
+                delta = service.data[ATTR_REMOVE_ENTITIES]
+                entity_ids = set(group.tracking)
+                new_entity_ids = set()
+                for entity in entity_ids:
+                    if entity not in set(delta):
+                        new_entity_ids.add(entity)
+                await group.async_update_tracked_entity_ids(new_entity_ids)
```

Comment:
Alternate implementation approach. This removes all items in the second set from the first set and returns the result

```suggestion
                new_entity_ids = set(group.tracking) - set(service.data[ATTR_REMOVE_ENTITIES])
                await group.async_update_tracked_entity_ids(new_entity_ids)
```

Reply:
You could also do xor since the add entities uses or (`|`)
```suggestion
new_entity_ids = set(group.tracking) ^ set(service.data[ATTR_REMOVE_ENTITIES])
                await group.async_update_tracked_entity_ids(new_entity_ids)
```

Reply:
That's how it goes when overthinking it. I'll change this